core/src/token_info/cached.rs

```diff
@@ -75,8 +74,22 @@ impl TokenInfoCache {
         ids.into_iter()
             .copied()
             .filter(|id| !cache.contains_key(id))
+            // NOTE: Make sure to `collect` to not hold the `cache` lock.
```

Comment:
Wow, that seems like a nasty one (I assume you were trying to return an iterator and ran in some kind of lock?). Should we maybe clone the cache to before iterating?

Reply:
I'm OK with just the comment here. A unit test caught the issue so it appears we are covered :smile: